tests/cpp/dlr_data_transform_test.cc

```diff
@@ -13,40 +13,48 @@ int main(int argc, char** argv) {
   return RUN_ALL_TESTS();
 }
 
+void ExpectFloatEq(float value, float expected) {
+  // TODO(trevmorr): NanSensitiveFloatEq from googlemock would be useful here.
+  if (std::isnan(expected)) {
+    EXPECT_TRUE(std::isnan(value));
+  } else {
+    EXPECT_EQ(value, expected);
+  }
+}
+
 TEST(DLR, DataTransformCategoricalString) {
   dlr::DataTransform transform;
   nlohmann::json metadata = R"(
     {
       "DataTransform": {
         "Input": {
-          "0": {
-            "CategoricalString": [
-              { "apple": 0, "banana": 1, "7": 2 }
-            ]
-          }
+          "ColumnTransform": [
+            {
+              "Type": "CategoricalString",
+              "Map": [{ "apple": 0, "banana": 1, "7": 2 }]
+            }
+          ]
         }
       }
     })"_json;
-  EXPECT_TRUE(transform.HasInputTransform(metadata, 0));
-  EXPECT_FALSE(transform.HasInputTransform(metadata, 1));
+  EXPECT_TRUE(transform.HasInputTransform(metadata));
 
   // User input
   const char* data = R"([["apple"], ["banana"], ["7"], [7], ["walrus"], [-5]])";
   std::vector<int64_t> shape = {static_cast<int64_t>(std::strlen(data))};
   // Model input
-  DLDataType dtype = DLDataType{kDLFloat, 32, 1};
+  std::vector<DLDataType> dtypes = {DLDataType{kDLFloat, 32, 1}};
   DLContext ctx = DLContext{kDLCPU, 0};
-  tvm::runtime::NDArray transformed_data;
-  EXPECT_NO_THROW(transformed_data =
-                      transform.TransformInput(metadata, 0, shape.data(), const_cast<char*>(data),
-                                               shape.size(), dtype, ctx));
+  std::vector<tvm::runtime::NDArray> transformed_data(1);
+  EXPECT_NO_THROW(transform.TransformInput(metadata, shape.data(), const_cast<char*>(data),
+                                           shape.size(), dtypes, ctx, &transformed_data));
 
   std::vector<float> expected_output = {0, 1, 2, -1, -1, -1};
-  EXPECT_EQ(transformed_data->ndim, 2);
-  EXPECT_EQ(transformed_data->shape[0], 6);
-  EXPECT_EQ(transformed_data->shape[1], 1);
+  EXPECT_EQ(transformed_data[0]->ndim, 2);
+  EXPECT_EQ(transformed_data[0]->shape[0], 6);
+  EXPECT_EQ(transformed_data[0]->shape[1], 1);
   for (size_t i = 0; i < expected_output.size(); ++i) {
-    CHECK_EQ(static_cast<float*>(transformed_data->data)[i], expected_output[i])
+    CHECK_EQ(static_cast<float*>(transformed_data[0]->data)[i], expected_output[i])
```

Comment:
I think it should be `gtest` method `EXPECT_EQ` here instead of TVM/dmlc `CHECK_EQ`. Otherwise `release-1.10.0` compilation fails (on Linux, not on Mac) with the following error
```
[100%] Building CXX object CMakeFiles/dlr_data_transform_test.dir/tests/cpp/dlr_data_transform_test.cc.o
[100%] Linking CXX executable dlr_data_transform_test
CMakeFiles/dlr_data_transform_test.dir/tests/cpp/dlr_data_transform_test.cc.o: In function `DLR_DataTransformCategoricalString_Test::TestBody()':
dlr_data_transform_test.cc:(.text+0x80a7): undefined reference to `tvm::runtime::detail::LogFatal::GetEntry()'
dlr_data_transform_test.cc:(.text+0x80fa): undefined reference to `tvm::runtime::detail::LogFatal::GetEntry()'
dlr_data_transform_test.cc:(.text+0x86df): undefined reference to `tvm::runtime::detail::LogFatal::GetEntry()'
CMakeFiles/dlr_data_transform_test.dir/tests/cpp/dlr_data_transform_test.cc.o: In function `tvm::runtime::detail::LogFatal::LogFatal(char const*, int)':
dlr_data_transform_test.cc:(.text._ZN3tvm7runtime6detail8LogFatalC2EPKci[_ZN3tvm7runtime6detail8LogFatalC5EPKci]+0x26): undefined reference to `tvm::runtime::detail::LogFatal::GetEntry()'
CMakeFiles/dlr_data_transform_test.dir/tests/cpp/dlr_data_transform_test.cc.o: In function `tvm::runtime::detail::LogFatal::Entry::Finalize()':
dlr_data_transform_test.cc:(.text._ZN3tvm7runtime6detail8LogFatal5Entry8FinalizeEv[_ZN3tvm7runtime6detail8LogFatal5Entry8FinalizeEv]+0x3a): undefined reference to `tvm::runtime::Backtrace[abi:cxx11]()'
collect2: error: ld returned 1 exit status
CMakeFiles/dlr_data_transform_test.dir/build.make:99: recipe for target 'dlr_data_transform_test' failed
make[2]: *** [dlr_data_transform_test] Error 1
CMakeFiles/Makefile2:702: recipe for target 'CMakeFiles/dlr_data_transform_test.dir/all' failed
make[1]: *** [CMakeFiles/dlr_data_transform_test.dir/all] Error 2
Makefile:145: recipe for target 'all' failed
make: *** [all] Error 2
```